mantidproject/mantid
# 25076 Add algorithm for clipping peaks

**Description of work.**

Implements the SNAPBackgroundPeakClipping algorithm provided in the referenced issue as a new algorithm 'ClipPeaks' which can be used to remove peaks from data. The majority of the algorithm already existed in 'SNAPReduce,' so the functions in the original file were updated to the newer ones found in SNAPReduce. The SNAPReduce algorithm will be refactored to call this new algorithm in a separate issue/pull request.

Documentation for the algorithm was added, including the equations the algorithm is based on. Some basic unit tests were created from adding peaks to a simple baseline function and verifying the ClipPeaks algorithm removed the peaks and was able to recreate a close version of the baseline function.

**To test:**

Download the attached script and open in Mantid: [clippeak_test.py.txt](https://github.com/mantidproject/mantid/files/5220415/clippeak_test.py.txt)

Once the script is executed, several workspaces will be created:
The "BaselineData" workspace can be plotted to show what the original function is.
"NoiseData" is the baseline function with randomly generated peaks added.

The tester should plot the spectrum of "PeakClippedData" and verify that the peaks randomly introduced in "NoiseData" are gone, and that it resembles the original "BaselineData" plot. 

The tester can also re-run the script several times, verifying the above result remains true for different random seeds.


Resolves #25076.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
